keps/sig-node/4960-container-stop-signals/README.md

```diff
@@ -0,0 +1,382 @@
+			stopSignal = container.StopSignal
+		} else {
+// rest of the code...
+```
+
+Obviously these are not all of the required changes, but a quick idea of what would need to be done. Find the entire diff for containerd [here](https://github.com/containerd/containerd/compare/main...sreeram-venkitesh:containerd:added-custom-stop-signal?expand=1).
+### User Stories (Optional)
+
+#### Story 1
+
+Kubernetes by default sends a SIGTERM to all containers while killing them. When running nginx on Kubernetes, this can result in nginx dropping requests as reported [here](https://github.com/Kong/kubernetes-ingress-controller/pull/283). The current solution for this issue would be to build custom images with a SIGQUIT stop signal or to write a PreStop lifecycle hook that manually kills the process gracefully, which is what is done in the PR. If we had stop signal support at the Container spec level, this would've been easier and straightforward to implement. Users wouldn't have to patch the applications running on Kubernetes to handle different termination behavior. This is also similar to [this issue](https://github.com/github/resque/pull/21). 
+
+### Risks and Mitigations
+
+I don't see any issues with adding a new field for the stop signal because this is an optional feature that users.
+
+## Design Details
+
+On top of the details described in the [Proposal](#proposal), these are some details on how exactly the new field will work.
+- `StopSignal` in the Container spec can be a nil value. In this case, the default stop signal of SIGTERM will be used.
+- The `StopSignal` field will override the stop signal set from the container image definition.
+- The order of priority for the different stop signals would look like this
+	`StopSignal from Container Spec > STOPSIGNAL from container image > Default value of SIGTERM`
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+Alpha:
+- Test that the validation fails when given a non string value for StopSignal
+- Test that the validation passes when given a proper string value representing a standard stop signal
+- Test that the validation fails when the StopSignal is given with the feature gate disabled
+- Test that the validation returns the appropriate error message when an invalid string value is given for StopSignal
+- Tests for verifying behavior when feature gate is disabled after being used to create Pods where StopSignal field is used
+- Tests for verifying behavior when feature gate is reenabled after being disabled after creating Pods with StopSignal
+
+##### Integration tests
+  
+##### e2e tests
+
+- Test that containers are killed with the right stop signal when StopSignal is passed
+- Test that containers are killed with SIGTERM when no StopSignal is passed
```

Comment:
Given that it we don't know when containerd/CRI-O would implement this, I would maybe suggest a test or verify the behavior on what happens if the StopSignal is specifed but the containerd runtime does not have this implementation.

Reply:
Added here: https://github.com/kubernetes/enhancements/pull/5122/commits/6cc0e8c265faec2eb04d791df0eda79f6dc32db6#diff-d5b499e0c69891876287d8717d528b33185bd612fb348c60a35c3c9c14f8d389R193